pkg/controllermanager/controller/shoot/shoot_control_reconcile.go

```diff
@@ -278,15 +278,25 @@ func (c *Controller) runReconcileShootFlow(o *operation.Operation, operationType
 			Fn:           flow.TaskFn(botanist.HibernateControlPlane).RetryUntilTimeout(defaultInterval, 2*time.Minute).DoIf(o.Shoot.HibernationEnabled),
 			Dependencies: flow.NewTaskIDs(initializeShootClients, deploySeedMonitoring, deploySeedLogging, deployClusterAutoscaler),
 		})
-		deployExtensionResource = g.Add(flow.Task{
+		deployExtensionResources = g.Add(flow.Task{
 			Name:         "Deploying extension resources",
-			Fn:           flow.TaskFn(botanist.DeployExtensionResources).RetryUntilTimeout(defaultInterval, defaultTimeout),
+			Fn:           flow.TaskFn(botanist.DeployExtensionResources).RetryUntilTimeout(defaultInterval, defaultTimeout).SkipIf(o.Shoot.HibernationEnabled),
 			Dependencies: flow.NewTaskIDs(initializeShootClients),
 		})
 		_ = g.Add(flow.Task{
 			Name:         "Waiting until extension resources are ready",
-			Fn:           flow.TaskFn(botanist.WaitUntilExtensionResourcesReady),
-			Dependencies: flow.NewTaskIDs(deployExtensionResource),
+			Fn:           flow.TaskFn(botanist.WaitUntilExtensionResourcesReady).SkipIf(o.Shoot.HibernationEnabled),
+			Dependencies: flow.NewTaskIDs(deployExtensionResources),
+		})
+		deleteStaleExtensionResources = g.Add(flow.Task{
+			Name:         "Delete stale extension resources",
+			Fn:           flow.TaskFn(botanist.DeleteStaleExtensionResources).RetryUntilTimeout(defaultInterval, defaultTimeout).SkipIf(o.Shoot.HibernationEnabled),
```

Comment:
Again, why not always?

Reply:
Agreed! But you agree with only waiting for the deletion if the shoot is not hibernated, right?

Reply:
Hm, not really, why should we do that?

Reply:
Let's say the extension controller deploys manifests to the shoot (possibly via G-Resource-Manager), the created resources cannot be cleaned up while the shoot is hibernated. So the extension controller will not remove the `finalizer` but we'd continue waiting.

Reply:
Good point, thanks.. yes, well, we might be safer with keeping it as you suggested. We don't know how the extension controllers are implemented and not waiting during hibernation would not block us in such case (while still the resource is marked for deletion correctly in the system).
However, if an extension is implemented in the way you mentioned it will also not be easily possible to entirely remove it from the system. There will most probably be always at least one hibernated shoot, and then the corresponding `Extension` remains in the system (which will block our `ControllerInstallation` controller).